File: Axiom/Internal/Intermediate/ImLabel.cs
using Axiom.Internal.Frames;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace Axiom.Internal.Intermediate
{
    internal class ImLabel : ImStatement
    {
        public ImLabel(Label label)
        {
            Label = label;
        }

        public override void Dump(TextWriter tw, int ident)
        {
            Report.Dump(tw, ident, "LABEL(" + Label.Name +")");
        }

        public override ImSequence Linear()
        {
            var linear = new ImSequence();

            linear.Statements.Add(this);

            return linear;
        }

        public override bool Equals(object obj)
        {
            if (obj is ImLabel) {
                return ((ImLabel)obj).Label.Name == Label.Name;
            }

            return false;
        }

        public override int GetHashCode()
        {
            return Label.GetHashCode();
        }

        public Label Label { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/im/ImLabel.java
package com.moybl.axiom.im;

import com.moybl.axiom.frame.Label;

public class ImLabel implements ImStatement {

	private Label label;

	public ImLabel(Label label) {
		this.label = label;
	}

	@Override
	public boolean equals(Object obj) {
		if (obj == null || obj.getClass() != this.getClass()) {
			return false;
		}

		return label.equals(((ImLabel) obj).label);
	}

	@Override
	public int hashCode() {
		return label.hashCode();
	}

	@Override
	public void accept(ImVisitor visitor) {
		visitor.visit(this);
	}

	public Label getLabel() {
		return label;
	}

}
